tests/cloner_test.go

```diff
@@ -3308,15 +3323,26 @@ func EnableGcAndAnnotateLegacyDv(f *framework.Framework, dvName, dvNamespace str
 	Expect(ok).To(BeFalse())
 
 	By("Add empty DeleteAfterCompletion annotation to DV for reconcile")
-	controller.AddAnnotation(dv, controller.AnnDeleteAfterCompletion, "")
-	dv, err = f.CdiClient.CdiV1beta1().DataVolumes(dvNamespace).Update(context.TODO(), dv, metav1.UpdateOptions{})
-	Expect(err).ToNot(HaveOccurred())
+	Eventually(func() error {
+		dv, err = f.CdiClient.CdiV1beta1().DataVolumes(dvNamespace).Get(context.TODO(), dvName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		controller.AddAnnotation(dv, controller.AnnDeleteAfterCompletion, "")
+		// We shouldn't make the test fail if there's a conflict with the update request.
+		// These errors are usually transient and should be fixed in subsequent retries.
+		dv, err = f.CdiClient.CdiV1beta1().DataVolumes(dvNamespace).Update(context.TODO(), dv, metav1.UpdateOptions{})
+		return err
+	}, timeout, pollingInterval).Should(Succeed())
+
 	VerifyNoGC(f, dvName, dvNamespace)
 
 	By("Add true DeleteAfterCompletion annotation to DV")
 	controller.AddAnnotation(dv, controller.AnnDeleteAfterCompletion, "true")
-	_, err = f.CdiClient.CdiV1beta1().DataVolumes(dvNamespace).Update(context.TODO(), dv, metav1.UpdateOptions{})
-	Expect(err).ToNot(HaveOccurred())
+
+	Eventually(func() error {
+		dv, err = f.CdiClient.CdiV1beta1().DataVolumes(dvNamespace).Update(context.TODO(), dv, metav1.UpdateOptions{})
```

Comment:
we don't refresh the datavolume here, so it'll probably keep failing if a conflict occurs
does this test fail often with a conflict err?

Reply:
Oops my fault again, It doesn't fail but yeah, I should make sure to refresh it.